worker/modelcache/worker.go

```diff
@@ -147,13 +147,20 @@ func (c *cacheWorker) loop() error {
 				// Continue through.
 			}
 			allWatcherStarts.Inc()
+			c.controller.Mark()
 			watcher := pool.SystemState().WatchAllModels(pool)
 			c.watcher = watcher
 			c.mu.Unlock()
 
 			err := c.processWatcher(watcher, watcherChanges)
 			if err == nil {
 				// We are done, so exit
+				result := c.controller.Sweep()
```

Comment:
Is this in the right place? It looks like we could mark things as stale that never get swept as long as the new watcher keeps running.

I think we instead need a flag set when we mark that is checked in the final `for` loop to cause a sweep then flag reset.

Reply:
I'll take another look.